@here/olp-sdk-dataservice-write/lib/client/VersionedLayerClient.ts

```diff
@@ -273,7 +275,50 @@ export class VersionedLayerClient {
         }
 
         if (!metadata.dataHandle) {
-            metadata.dataHandle = Uuid.create();
+            /**
+             * If datahandle was not provided, we generate the UUID and use it as datahandle.
+             * A retry logic in case datahandle is already present is a loop of 3 tries
+             * and each time we generate a new UUID and retry.
+             *
+             * If all thies was not success, the method rejects the promise with Error.
+             */
+            const timesToGenerate = 3;
+            const checkDataExistsRequest = new CheckDataExistsRequest().withLayerId(
+                layerId
+            );
+            if (billingTag) {
+                checkDataExistsRequest.withBillingTag(billingTag);
+            }
+
+            for (
+                let generatingCount = 0;
+                generatingCount < timesToGenerate;
+                generatingCount++
+            ) {
+                const generatedDatahandle = Uuid.create();
+                const isDataExist = await this.checkDataExists(
+                    checkDataExistsRequest.withDataHandle(generatedDatahandle)
+                ).catch(async (response: Response) => {
+                    if (response.status === 404) {
```

Comment:
Can you add 404 to the STATUS_CODES enum and used the enum?

Reply:
Made it public and moved to the core and use, thanks.